wagtail/wagtailadmin/views/pages.py

```diff
@@ -588,16 +588,27 @@ def unpublish(request, page_id):
         raise PermissionDenied
 
     if request.method == 'POST':
+        include_descendants = request.POST.get("include_descendants", False)
+
         page.unpublish()
 
+        if include_descendants == 'True':
+            descendant_pages = page.get_descendants().live()
+            for descendant in descendant_pages:
+                descendant = descendant.specific
+                if descendant.permissions_for_user(request.user).can_unpublish():
```

Comment:
As permissions in Wagtail can't be taken away as you go deeper in the tree, this check should never fail